third_party/move/move-compiler-v2/tests/reference-safety/freeze_mutable.move

```diff
@@ -0,0 +1,12 @@
+module 0x8675309::Tester {
+
+    fun foo(_other: &mut u64): &mut u64 {
+        _other
+    }
+
+    fun test(result: &mut u64): &u64 {
+        let returned_ref = foo(result);
+        freeze(result);
```

Comment:
Could we add an example where `returned_ref` is not used later, like here? In such a case, we should not give an error, right?

Reply:
@rahxephon89 Just checking if this comment was addressed.

Reply:
The PR was unfortunately automatically merged but this PR shows that no error is generated: https://github.com/aptos-labs/aptos-core/pull/14248